### PR TITLE
Added support for AIX wpars

### PR DESCRIPTION
**Pull request by scotthain**
90% of this is work done by @curiositycasualty - I just redid the branch and made this PR after a few minor tweaks.

@chef-cookbooks/engineering-services

### CODE REVIEW

File: libraries/remote_install.rb [hunk 1]
```diff
@@ -110,7 +110,8 @@ def verify
     end
 
     def extract
-      extract_command = 'tar -xv'
+      # None of AIX's coreutils like '-v' ¯\_(ツ)_/¯
+      extract_command = (aix? ? 'tar -x' : 'tar -xv')
```
Comment (schisamo) [inline libraries/remote_install.rb]: Do we need the special case? Can't we just use `tar -x` for all platforms?

File: recipes/_user.rb [hunk 2]
```diff
@@ -35,13 +35,23 @@
...
   home     build_user_home
   supports manage_home: true
   password node['omnibus']['build_user_password']
   unless windows?
-    shell '/usr/local/bin/bash'
-    gid   node['omnibus']['build_user_group']
+    if aix?
+    shell node['omnibus']['build_user_shell']
+    else
+      shell '/usr/local/bin/bash'
```
Comment (schisamo) [inline recipes/_user.rb]: We should move this logic to the attributes file where `node['omnibus']['build_user_shell']` is defined.

Comment (scotthain) [inline recipes/_user.rb]: Works for me - that code felt really awkward and I couldn't think offhand of a better way.

**Comment by scotthain** [comment]
@schisamo I think we're good now

File: recipes/_omnibus_toolchain.rb [hunk 3]
```diff
@@ -48,15 +48,27 @@
...
+remote_file package_path do
+  source package_url
+  action :create_if_missing
+  not_if { installed }
+end
+
+package node['omnibus']['toolchain_name'] do
+  source package_path
+  options install_options
+  not_if { installed }
```
Comment (schisamo) [inline recipes/_omnibus_toolchain.rb]: Has this code been tested? Pretty sure Ruby is going to get cranky if you call the variable `installed` and it hasn't been defined (basically all non-AIX platforms and AIX when the package does not exist).

Comment (scotthain) [inline recipes/_omnibus_toolchain.rb]: Hmmm yeah it has been tested but only on AIX (where it does work in both cases). @curiositycasualty can you take a peek at this please?

**Pull request closed without merging**